crates/workspace/src/workspace.rs

```diff
@@ -1259,21 +1261,31 @@ impl Workspace {
     }
 
     fn send_keystrokes(&mut self, action: &SendKeystrokes, cx: &mut ViewContext<Self>) {
-        let keystrokes: Vec<Keystroke> = action
+        let mut keystrokes: Vec<Keystroke> = action
             .0
             .split(" ")
             .flat_map(|k| Keystroke::parse(k).log_err())
             .collect();
-
-        cx.window_context().defer(move |cx| {
-            for keystroke in keystrokes {
-                cx.dispatch_event(PlatformInput::KeyDown(KeyDownEvent {
-                    keystroke: keystroke.clone(),
-                    is_held: false,
-                }));
-                cx.dispatch_event(PlatformInput::KeyUp(KeyUpEvent { keystroke }));
-            }
-        })
+        keystrokes.reverse();
+
+        self.dispatching_keystrokes
```

Comment:
What's the purpose of this `dispatching_keystrokes` field? It looks like the parsed keystrokes are appended to this field, but I don't see where the field is being used outside of this.

Reply:
It allows you to include keys that themselves are bound to `SendKeystrokes`, so you can call this recursively.

In order to do that reliably we need to make sure that we append the new keystrokes to the existing list to be sure they'll be processed in order.